src/Utils.ts

```diff
@@ -1,29 +1,65 @@
-import * as fs from "fs-extra";
+import * as fse from "fs-extra";
 import * as http from "http";
 import * as md5 from "md5";
 import * as minimatch from "minimatch";
 import * as os from "os";
 import * as path from "path";
-import { extensions, workspace } from 'vscode';
+import { ExtensionContext, extensions, workspace } from 'vscode';
 import * as xml2js from "xml2js";
 import { Archetype } from "./model/Archetype";
 import { ProjectItem } from "./model/ProjectItem";
 import { IArchetype, IArchetypeCatalogRoot, IArchetypes, IPomRoot } from "./model/XmlSchema";
-const EXTENSION_NAME: string = "vscode-maven";
-const EXTENSION_ID: string = "eskibear.vscode-maven";
+
+let EXTENSION_PUBLISHER: string;
+let EXTENSION_NAME: string;
+let EXTENSION_VERSION: string;
+let EXTENSION_AI_KEY: string;
 
 export namespace Utils {
-    export async function getProject(absolutePath: string, workspacePath: string, iconPath?: string): Promise<ProjectItem> {
-        if (await fs.pathExists(absolutePath)) {
-            const xml: string = await fs.readFile(absolutePath, "utf8");
+    export function loadPackageInfo(context: ExtensionContext): void {
+        const { publisher, name, version, aiKey } = fse.readJSONSync(context.asAbsolutePath("./package.json"));
+        EXTENSION_AI_KEY = aiKey;
+        EXTENSION_PUBLISHER = publisher;
+        EXTENSION_NAME = name;
+        EXTENSION_VERSION = version;
+    }
+
+    export function getExtensionPublisher(): string {
+        return EXTENSION_PUBLISHER;
+    }
+
+    export function getExtensionName(): string {
+        return EXTENSION_NAME;
+    }
+
+    export function getExtensionId(): string {
+        return `${EXTENSION_PUBLISHER}.${EXTENSION_NAME}`;
+    }
+
+    export function getExtensionVersion(): string {
+        return EXTENSION_VERSION;
+    }
+
+    export function getAiKey(): string {
+        return EXTENSION_AI_KEY;
+    }
+
+    export function getTempFolderPath(...args: string[]): string {
+        return path.join(os.tmpdir(), getExtensionId(), ...args);
+    }
+
+    export function getExtensionRootPath(...args: string[]): string {
+        return path.join(extensions.getExtension(getExtensionId()).extensionPath, ...args);
```

Comment:
how about passing the context as an argument and call `context.extensionPath`

Reply:
`context` cannot be accessed elsewhere, I think it's better to get extension path by id